lib/lambda_ethereum_consensus/beacon/pending_blocks.ex

```diff
@@ -114,65 +112,45 @@ defmodule LambdaEthereumConsensus.Beacon.PendingBlocks do
     end)
   end
 
-  @empty_mapset MapSet.new()
-
-  @impl true
-  def handle_info(:download_blocks, %{blocks_to_download: to_download} = state)
-      when to_download == @empty_mapset do
-    schedule_blocks_download()
-    {:noreply, state}
-  end
-
   @impl true
   def handle_info(:download_blocks, state) do
-    blocks_in_store = state.blocks_to_download |> MapSet.filter(&ForkChoice.has_block?/1)
+    blocks_to_download = state |> Map.filter(fn {_, {_, s}} -> s == :download end) |> Map.keys()
 
     downloaded_blocks =
-      state.blocks_to_download
-      |> MapSet.difference(blocks_in_store)
-      |> Enum.take(16)
-      |> BlockDownloader.request_blocks_by_root()
-      |> case do
-        {:ok, signed_blocks} ->
-          signed_blocks
-
-        {:error, reason} ->
-          Logger.debug("Block download failed: '#{reason}'")
-          []
+      if Enum.empty?(blocks_to_download) do
+        []
+      else
```

Comment:
nit and optional: maybe the empty case should be automatically handled by the block downloader with a 

```elixir
BlockDownloader.request_blocks_by_root([]), do: []
```

It's nicer to use 

```elixir
blocks_to_download
|> Enum.take(16)
|> BlockDownloader.request_blocks_by_root()
#...
```

directly and it makes the downloader module more robust.